spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -116,15 +117,15 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
     val relativizeIgnoreError =
       spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_VACUUM_RELATIVIZE_IGNORE_ERROR)
 
+    val canonicalizedBasePath = SparkPath.fromPathString(basePath).urlEncoded
     snapshot.stateDS.mapPartitions { actions =>
       val reservoirBase = new Path(basePath)
       val fs = reservoirBase.getFileSystem(hadoopConf.value.value)
       actions.flatMap {
         _.unwrap match {
-          case fa: FileAction if checkAbsolutePathOnly && !fa.path.contains(basePath) =>
-            Nil
-          case tombstone: RemoveFile if tombstone.delTimestamp < deleteBeforeTimestamp =>
-            Nil
+          case fa: FileAction if checkAbsolutePathOnly &&
+            !fa.path.contains(canonicalizedBasePath) => Nil
```

Comment:
So if a table is currently incorrectly storing its path with non-URL encoded escaping, does that mean we will accidentally delete all the files in VACUUM? But I guess the whole table would be messed up already in that case.

Reply:
Good point!
Let's check both `basePath` and `canonicalizedBasePath` to keep it safe.

Reply:
As an encoding unrelated side note: Shouldn't this be (have been) `fa.path.startsWith(canonicalizedBasePath)` and not `fa.path.contains(canonicalizedBasePath)`? I assume `startsWith` is a much cheaper check and it'd be very weird if the basepath occurred anywhere else...

Reply:
I don't know, will check.
If `contains` is necessary, I would guess it's likely to accommodate the case where there exists a scheme (e.g. `s3:/`) or not.

Reply:
I checked the origin of involved variables. It seems to me that `startsWith` will work because all paths are canonicalized by the URI class. Now I'm running tests just to confirm.